File: py/src/verification_sdk/types/attestation_model.py
from __future__ import annotations

from pydantic import BaseModel

from .attestation_common import TcbInfo, SigningAlgo


class ModelAttestation(BaseModel):
    request_nonce: str
    signing_algo: SigningAlgo
    signing_address: str
    intel_quote: str
    nvidia_payload: str
    info: ModelInfo


class ModelInfo(BaseModel):
    tcb_info: TcbInfo | str


class ModelAttestationReport(BaseModel):
    all_attestations: list[ModelAttestation]


class VerifyModelAttestationConfig(BaseModel):
    image_names_of_sigstore_hash: list[str]

File: rs/src/types/attestation_model.rs
use crate::types::attestation_common::{SigningAlgo, TcbInfoOrRaw};
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct ModelAttestation {
    pub request_nonce: String,
    pub signing_algo: SigningAlgo,
    pub signing_address: String,
    pub intel_quote: String,
    pub nvidia_payload: String,
    pub info: ModelAttestationInfo,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct ModelAttestationInfo {
    pub tcb_info: TcbInfoOrRaw,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct ModelAttestationReport {
    pub all_attestations: Vec<ModelAttestation>,
}
